providers/vsphere/provider/provider.go

```diff
@@ -0,0 +1,218 @@
+	}
+}
+
+func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error) {
+	flags := req.Flags
+	if flags == nil {
+		flags = map[string]*llx.Primitive{}
+	}
+
+	conn := &inventory.Config{
+		Type: req.Connector,
+	}
+
+	// parse args
+	user := ""
+	port := 0
+	if len(req.Args) != 0 {
+		target := req.Args[0]
+		if !strings.Contains(target, "://") {
+			target = "ssh://" + target
+		}
+
+		x, err := url.Parse(target)
+		if err != nil {
+			return nil, errors.New("incorrect format of target, please use user@host:port")
+		}
+
+		user = x.User.Username()
+		conn.Host = x.Hostname()
+		conn.Path = x.Path
+
+		if sPort := x.Port(); sPort != "" {
+			port, err = strconv.Atoi(x.Port())
+			if err != nil {
+				return nil, errors.New("port '" + x.Port() + "'is incorrectly formatted, must be a number")
+			}
+		}
+	}
+
+	if port > 0 {
+		conn.Port = int32(port)
+	}
+
+	// parse flags
+	if x, ok := flags["password"]; ok && len(x.Value) != 0 {
+		conn.Credentials = append(conn.Credentials, vault.NewPasswordCredential(user, string(x.Value)))
+	}
+
+	asset := inventory.Asset{
+		Connections: []*inventory.Config{conn},
+	}
+
+	return &plugin.ParseCLIRes{Asset: &asset}, nil
+}
+
+func (s *Service) Connect(req *plugin.ConnectReq, callback plugin.ProviderCallback) (*plugin.ConnectRes, error) {
+	if req == nil || req.Asset == nil {
+		return nil, errors.New("no connection data provided")
+	}
+
+	conn, err := s.connect(req, callback)
+	if err != nil {
+		return nil, err
+	}
+
+	// We only need to run the detection step when we don't have any asset information yet.
+	if req.Asset.Platform == nil {
+		if err := s.detect(req.Asset, conn); err != nil {
+			return nil, err
+		}
+	}
+
+	return &plugin.ConnectRes{
+		Id:        conn.ID(),
+		Name:      conn.Name(),
+		Asset:     req.Asset,
+		Inventory: nil,
+	}, nil
+}
+
+func (s *Service) connect(req *plugin.ConnectReq, callback plugin.ProviderCallback) (*connection.VsphereConnection, error) {
+	if len(req.Asset.Connections) == 0 {
+		return nil, errors.New("no connection options for asset")
+	}
+
+	asset := req.Asset
+	conf := asset.Connections[0]
+	var conn *connection.VsphereConnection
+	var err error
+
+	switch conf.Type {
+	default:
+		s.lastConnectionID++
+		conn, err = connection.NewVsphereConnection(s.lastConnectionID, asset, conf)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	var upstream *upstream.UpstreamClient
+	if req.Upstream != nil {
+		upstream, err = req.Upstream.InitClient()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	asset.Connections[0].Id = conn.ID()
+	s.runtimes[conn.ID()] = &plugin.Runtime{
+		Connection:     conn,
+		Resources:      map[string]plugin.Resource{},
+		Callback:       callback,
+		HasRecording:   req.HasRecording,
+		CreateResource: resources.CreateResource,
+		Upstream:       upstream,
+	}
+
+	return conn, err
+}
+
+func (s *Service) detect(asset *inventory.Asset, conn *connection.VsphereConnection) error {
+	// TODO: adjust asset detection
```

Comment:
Is this comment still needed?

Reply:
This still needs to be updated with the full discovery, which is not implemented in this PR